internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/controller/TestControllerKMERegistrationFromMessageHeader.java

```diff
@@ -0,0 +1,131 @@
+  }
+
+  @Test(timeOut = TEST_TIMEOUT)
+  public void testKMERegistrationThroughAdminTopicChannel() {
+    String clusterName = CLUSTER_NAMES[0];
+    String storeName = Utils.getUniqueString("store");
+
+    VeniceControllerWrapper pController =
+        multiRegionMultiClusterWrapper.getLeaderParentControllerWithRetries(clusterName);
+
+    VeniceMultiClusterWrapper child = childDatacenters.get(0);
+
+    // Remove the latest schema from child controller's local value serializer and remove it from child colo's schema
+    // repository (ZK).
+    VeniceControllerWrapper leaderController = child.getLeaderController(clusterName);
+    KafkaValueSerializer valueSerializer =
+        leaderController.getController().getVeniceControllerService().getKafkaValueSerializer();
+    TestUtils.waitForNonDeterministicAssertion(30, TimeUnit.SECONDS, true, () -> {
+      Assert.assertTrue(
+          valueSerializer.getProtocolVersionSize() >= AvroProtocolDefinition.KAFKA_MESSAGE_ENVELOPE
+              .getCurrentProtocolVersion());
+    });
+    HelixReadWriteSchemaRepositoryAdapter adapter =
+        (HelixReadWriteSchemaRepositoryAdapter) (leaderController.getVeniceHelixAdmin()
+            .getHelixVeniceClusterResources(clusterName)
+            .getSchemaRepository());
+    HelixReadWriteSchemaRepository repo =
+        (HelixReadWriteSchemaRepository) adapter.getReadWriteRegularStoreSchemaRepository();
+    repo.removeValueSchema(
+        AvroProtocolDefinition.KAFKA_MESSAGE_ENVELOPE.getSystemStoreName(),
+        AvroProtocolDefinition.KAFKA_MESSAGE_ENVELOPE.getCurrentProtocolVersion());
+    valueSerializer.removeAllSchemas();
+    LOGGER.info("all schemas are removed");
+
+    // Verify that the latest version of the protocol is deleted in ZK.
+    TestUtils.waitForNonDeterministicAssertion(30, TimeUnit.SECONDS, true, () -> {
+      Assert.assertTrue(
+          repo.getValueSchema(
+              AvroProtocolDefinition.KAFKA_MESSAGE_ENVELOPE.getSystemStoreName(),
+              AvroProtocolDefinition.KAFKA_MESSAGE_ENVELOPE.getCurrentProtocolVersion()) == null);
+    });
+
+    /*
+     * Calling parent controller's create store action which will trigger an admin message which contains
+     * the latest schema in its header, child controller when it encounters the new schema in the message header,
+     * would register the new schema into the child colo's schema repo as well as add to its local serializer.
+     */
+
+    pController.getVeniceAdmin().createStore(clusterName, storeName, "", "\"string\"", "\"string\"", false);
+
+    // Verify that schema is registered in the child colo's schema repo.
+    TestUtils.waitForNonDeterministicAssertion(30, TimeUnit.SECONDS, true, () -> {
+      Assert.assertTrue(
+          repo.getValueSchema(
+              AvroProtocolDefinition.KAFKA_MESSAGE_ENVELOPE.getSystemStoreName(),
+              AvroProtocolDefinition.KAFKA_MESSAGE_ENVELOPE.getCurrentProtocolVersion()) != null);
```

Comment:
This doesn't seem strict enough?  Could we get the protocol version that we expect at the top of the test?

Reply:
Yeah, we can add the same check in the beginning of the test.

Reply:
!= null part, I think we should check the version number

Reply:
okay, sure, will check that.